src/Adapter/Currency/CommandHandler/UpdateLiveExchangeRatesHandler.php

```diff
@@ -0,0 +1,252 @@
+use PrestaShop\PrestaShop\Core\Domain\Currency\CommandHandler\UpdateLiveExchangeRatesHandlerInterface;
+use PrestaShop\PrestaShop\Core\Domain\Currency\Exception\CannotCreateLiveExchangeUpdateCronTaskException;
+use PrestaShop\PrestaShop\Core\Domain\Currency\Exception\CurrencyException;
+use PrestaShop\PrestaShop\Core\Domain\Currency\Exception\DisabledLiveExchangeRatesException;
+use PrestaShopException;
+use Shop;
+use Symfony\Component\Translation\TranslatorInterface;
+
+/**
+ * Class UpdateLiveExchangeRatesHandler is responsible for turning on or off the setting - if its on then
+ * in CronJobs module it creates new record with url which points to the script which is being executed at certain time
+ * of period. If the setting is off then it removes that record.
+ *
+ * @internal
+ */
+final class UpdateLiveExchangeRatesHandler implements UpdateLiveExchangeRatesHandlerInterface
+{
+    /**
+     * @var Configuration
+     */
+    private $configuration;
+
+    /**
+     * @var Tools
+     */
+    private $tools;
+
+    /**
+     * @var Shop
+     */
+    private $contextShop;
+
+    /**
+     * @var bool
+     */
+    private $isCronJobModuleInstalled;
+
+    /**
+     * @var ShopUrlDataProvider
+     */
+    private $shopUrlDataProvider;
+
+    /**
+     * @var string
+     */
+    private $adminBaseUrl;
+
+    /**
+     * @var string
+     */
+    private $dbPrefix;
+
+    /**
+     * @var TranslatorInterface
+     */
+    private $translator;
+
+    /**
+     * @param Configuration $configuration
+     * @param Tools $tools
+     * @param Shop $contextShop
+     * @param ShopUrlDataProvider $shopUrlDataProvider
+     * @param TranslatorInterface $translator
+     * @param bool $isCronJobModuleInstalled
+     * @param string $adminBaseUrl
+     * @param string $dbPrefix
+     */
+    public function __construct(
+        Configuration $configuration,
+        Tools $tools,
+        Shop $contextShop,
+        ShopUrlDataProvider $shopUrlDataProvider,
+        TranslatorInterface $translator,
+        $isCronJobModuleInstalled,
+        $adminBaseUrl,
+        $dbPrefix
+    ) {
+        $this->configuration = $configuration;
+        $this->tools = $tools;
+        $this->contextShop = $contextShop;
+        $this->isCronJobModuleInstalled = $isCronJobModuleInstalled;
+        $this->shopUrlDataProvider = $shopUrlDataProvider;
+        $this->adminBaseUrl = $adminBaseUrl;
+        $this->dbPrefix = $dbPrefix;
+        $this->translator = $translator;
+    }
+
+    /**
+     * {@inheritdoc}
+     *
+     * @throws CurrencyException
+     */
+    public function handle(UpdateLiveExchangeRatesCommand $command)
+    {
+        if (!$this->isCronJobModuleInstalled) {
+            throw new DisabledLiveExchangeRatesException(
+                'Live exchange rates feature cannot be modified due to cronjob module is uninstalled'
+            );
+        }
+
+        $this->configuration->restrictUpdatesTo($this->contextShop);
```

Comment:
This part is tricky:

I left exactly how it was on legacy currencies controller - it was restricted to certain shop context only. And yes, on "All shops context" it works in unexpected way.

However, the stuff it generates is the url which calls `cron_currency_rates.php` file which iterates over all shops anyway :o 

I guess it was made to be that way that the cron would be executed only on one shop :thinking: 

maybe here I need to pass `contextShopIds` which will be able to create cron urls for all shops - but this will result in same file call from all shop contexts.

Not sure how to do here. Any thoughts @matks , @sarjon  ?

Reply:
OK, we are dealing with a wild usecase here 😄 

I think we're going to leave it the way it is right now because modifying this behavior requires a lot of rework.

What should be done in this rework:
- it should be required to pass `cron_currency_rates.php` a shop ID so the script knows which shop rates need to be updated
- each cron entry should call the script passing it's own shop ID so a merchant can really control which shop gets updated and how often

This is what makes sense to me 🤔 

If it looks right to you, can you create an issue so we dont forget to do this (I'll put it in the Migration kanban) and add a `@todo` statement in the code ? This way we won't forget (when we have migrated all SF pages, we'll scan for all `@todo` statements and this should indicate what's left to do).